sphinxcontrib/needs/needs.py

```diff
@@ -1,148 +1,243 @@
+                title="Implementation",
+                prefix="I_",
+                color="#DF744A",
+                style="node",
+            ),
+            dict(
+                directive="test",
+                title="Test Case",
+                prefix="T_",
+                color="#DCB239",
+                style="node",
+            ),
+            # Kept for backwards compatibility
+            dict(
+                directive="need",
+                title="Need",
+                prefix="N_",
+                color="#9856a5",
+                style="node",
+            ),
+        ],
+        "html",
+    )
+    app.add_config_value("needs_include_needs", True, "html", types=[bool])
+    app.add_config_value("needs_need_name", "Need", "html", types=[str])
+    app.add_config_value("needs_spec_name", "Specification", "html", types=[str])
+    app.add_config_value("needs_id_prefix_needs", "", "html", types=[str])
+    app.add_config_value("needs_id_prefix_specs", "", "html", types=[str])
+    app.add_config_value("needs_id_length", 5, "html", types=[int])
+    app.add_config_value("needs_specs_show_needlist", False, "html", types=[bool])
+    app.add_config_value("needs_id_required", False, "html", types=[bool])
+    app.add_config_value(
+        "needs_id_regex",
+        "^[A-Z0-9_]{{{id_length},}}".format(id_length=app.config.needs_id_length),
+        "html",
+    )
+    app.add_config_value("needs_show_link_type", False, "html", types=[bool])
+    app.add_config_value("needs_show_link_title", False, "html", types=[bool])
+    app.add_config_value("needs_file", "needs.json", "html")
+    app.add_config_value(
+        "needs_table_columns", "ID;TITLE;STATUS;TYPE;OUTGOING;TAGS", "html"
+    )
+    app.add_config_value("needs_table_style", "DATATABLES", "html")
+
+    app.add_config_value("needs_role_need_template", u"{title} ({id})", "html")
+    app.add_config_value("needs_role_need_max_title_length", 30, "html", types=[int])
+
+    app.add_config_value("needs_extra_options", {}, "html")
+    app.add_config_value("needs_title_optional", False, "html", types=[bool])
+    app.add_config_value("needs_max_title_length", -1, "html", types=[int])
+    app.add_config_value("needs_title_from_content", False, "html", types=[bool])
+
+    app.add_config_value("needs_diagram_template", DEFAULT_DIAGRAM_TEMPLATE, "html")
+
+    app.add_config_value("needs_functions", [], "html", types=[list])
+    app.add_config_value("needs_global_options", {}, "html", types=[dict])
+
+    app.add_config_value("needs_duration_option", "duration", "html")
+    app.add_config_value("needs_completion_option", "completion", "html")
 
     # If given, only the defined status are allowed.
     # Values needed for each status:
     # * name
     # * description
     # Example: [{"name": "open", "description": "open status"}, {...}, {...}]
-    app.add_config_value('needs_statuses', [], 'html')
+    app.add_config_value("needs_statuses", [], "html")
```

Comment:
Is there any common rule why to prefer " ?
I normally use ' as it is easier to reach on german keyboard layout.
And " when I need to use ' in the string itself:
- ``"Hi, my name is 'foo' "``
or ``f"Hi, name name is {config['name']}"``

Just want to be sure that we have a reason why this gets changed.
If there is no reason, I personal would like to stay with ', as it makes thing easier.

Reply:
i'm using [Black](https://github.com/psf/black) to format these.

Black is a deliberately opinionated and unconfigurable formatter. It offers no way to configure the formatting of single quotes vs double quotes

see here - https://github.com/psf/black/blob/master/docs/the_black_code_style.md
and in particular here - https://github.com/psf/black/blob/master/docs/the_black_code_style.md#strings

> On certain keyboard layouts like US English, typing single quotes is a bit easier than double quotes. The latter requires use of the Shift key. My recommendation here is to keep using whatever is faster to type and let Black handle the transformation.

that said, i don't have a strong opinion, and Black is far from the only option.

I think regardless of the style that you choose @danwos, this project should absolutely recommend a formatting tool and provide a configuration (unless it's Black and there is no configuration!)

Reply:
maybe it's best to drop the formatting commit from this pull request and open a separate issue?

unfortunately that commit is pretty low in the stack, so i've created a pretty horrible rebase for myself, i suspect ;)